domain/src/test/java/org/oppia/android/domain/topic/TopicListControllerTest.kt

```diff
@@ -499,6 +499,64 @@ class TopicListControllerTest {
     )
   }
 
+  @Test
+  fun testGetStoryList_story0DonePlayStory1FirstTestTopic_PlaySecondTestTopic_storyListIsCorrect() {
+    storyProgressController.recordCompletedChapter(
+      profileId0,
+      TEST_TOPIC_ID_0,
+      TEST_STORY_ID_0,
+      TEST_EXPLORATION_ID_2,
+      getCurrentTimestamp()
+    )
+    testCoroutineDispatchers.runCurrent()
+
+    storyProgressController.recordCompletedChapter(
+      profileId0,
+      TEST_TOPIC_ID_0,
+      TEST_STORY_ID_0,
+      TEST_EXPLORATION_ID_5,
+      getCurrentTimestamp()
+    )
+
+    testCoroutineDispatchers.runCurrent()
+
+    storyProgressController.recordRecentlyPlayedChapter(
+      profileId0,
+      TEST_TOPIC_ID_0,
+      TEST_STORY_ID_1,
+      TEST_EXPLORATION_ID_1,
+      getCurrentTimestamp()
+    )
+    testCoroutineDispatchers.runCurrent()
+
+    storyProgressController.recordRecentlyPlayedChapter(
+      profileId0,
+      TEST_TOPIC_ID_1,
+      TEST_STORY_ID_2,
+      TEST_EXPLORATION_ID_4,
+      getCurrentTimestamp()
+    )
+    testCoroutineDispatchers.runCurrent()
+
+    val promotedActivityList = retrievePromotedActivityList()
```

Comment:
Below: we should also be verifying isTopicLearned is set to differentiate between the 2 kinds of recently played stories.

Reply:
I am not getting clear understanding on this..

Reply:
The tests need to also check when a recently played story corresponds to a topic that hasn't already had a finished story vs. one who does (since this changes how the final list of cards are sorted in the carousel in the UI).

Reply:
Was this completed?

Reply:
Ah, as discussed I think we should move the isLearned check outside of the verify methods and put them directly in tests (& all tests with ongoing promoted stories should have an isLearned verification for true/false).

Reply:
@veena14cs I think you might not have addressed my comment above correctly. Please move ``isTopicLearned`` into the test bodies, not in their helpers. The concern is that the helpers hide that particular attribute, and it's important to specifically make it clear in the tests whether the topic is considered learned.

Reply:
Added `isTopicLearned` inside testbodies.